### PR TITLE
CLOUDP-156527: add regression tests

### PR DESCRIPTION
**Pull request by wtrocki**
I am adding regression tests to detect possible breaking changes in the happy path. 
Those tests would be only temporary as we would have breaking changes detection done by versioned API

### CODE REVIEW

File: v2/test/regression/multi_cloud_clusters.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+		Key:   &_string,
+		Value: &_string,
+	}
+
+	instanceSize := mongodbatlas.INSTANCESIZE_M10
+
+	autoScale := &mongodbatlas.AutoScalingV15{
+		Compute: &mongodbatlas.ComputeAutoScalingV15{
+			Enabled: &_true,
+			// TODO missing oneOf transformation
```
Comment (gssbzn) [inline v2/test/regression/multi_cloud_clusters.go:22]: can you link a CLOUDP?

File: v2/test/regression/multi_cloud_clusters.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+import (
+	mongodbatlas "go.mongodb.org/atlas/mongodbatlas/v2/api"
+)
+
+// Backwards compatibility, regression tests for the multi cloud clusters
+func TestMultiCloudClusters_ModelRegression() {
+	_true := true
+	_string := ""
+	_int := int32(0)
```
[the remark this answers is not in the document]
Comment (gssbzn) [reply v2/test/regression/multi_cloud_clusters.go:11]: or with generics now you could implement the function yourself 
```go
func Pointer[T any](x T) *T {
	return &x
}
```

**Review by gssbzn — APPROVED** [review]
some small comments not blocking

File: v2/test/regression/multi_cloud_clusters.go (hunk 1 above)
Comment (wtrocki) [reply v2/test/regression/multi_cloud_clusters.go:22]: We addressed that already. Good catch

**Pull request merged**